src/Compiler/TypedTree/TypedTreePickle.fs

```diff
@@ -1949,7 +1949,7 @@ and p_tcaug p st =
       (p_space 1)
       (p.tcaug_compare,
        p.tcaug_compare_withc,
-       p.tcaug_hash_and_equals_withc,
+       p.tcaug_hash_and_equals_withc |> Option.map (fun (v1, v2, v3, _) -> (v1, v2, v3)),
```

Comment:
So, we explicitly don't want to pickle the new augmentation?

Reply:
In other words, if pickled data only has the old `Equals` augmentation:
```
Equals(obj, IEqualityComparer)
```
When we codegen? Will it chose the one from IL or one from pickled data? We probably should add test for it (IL test with two different dlls).

Reply:
AFAIU this is the same concern as Tomas has expressed elsewhere - if so then this is addressed and tested [here](Okay so this also works now thanks to pushing this overload to the ad-hoc table. Tests are here: https://github.com/dotnet/fsharp/pull/16857/files#diff-df081675edf7d8decef09032e5ea509176d734fa6cc175ca888bc6b965acbe98)